### DISCUSSION TITLE
Should the emphasis be on Profiling?

### DISCUSSION THREAD

**Discussion by ChadBailey** [jaegertracing/jaeger#3758] [General]
# The Concern

I am brand new to Jaeger and similar tools. The description is currently:

> Jaeger: open source, end-to-end distributed tracing

Before I dig into my concern here, I just want the reader to know I am _really_ not concerned about winning an argument or being overly pedantic, so please hear me out before you dismiss my concern.

As a newcomer to this world, I had no idea that the term "tracing" had taken on its own niche meaning and this caused a huge deal of confusion between myself and several coworkers. When the person describing this tool and what it does to us would use the term tracing, what we lacked was an understanding of what was meant by a trace which appears to be tribal knowledge of the observability world.

## What is "Tracing"?

Further compounding the issue, we all know what stack tracing, tracelogging, and in general understand the true meaning of what a [trace](https://en.wikipedia.org/wiki/Tracing_(software)) is in the more general context. So when we were being told "it gives you traces", we did not understand this seems to mean something different than what was actually being said.

Again I will re-emphasize that I'm new to all of this, so I may _still not have this right_. From what I have been able to see so far though, my fresh eyes think that all of this could have been avoided by characterizing this tool a little differently. In other words, we needed to know quickly without having to read the docs what service this tool actually provides.

## What is "Profiling"?

What I consider this tool to be is a distributed [profiling](https://en.wikipedia.org/wiki/Profiling_(computer_programming)) tool for microservices. The reason I would characterize it as chiefly a profiling tool is at the end of the day that is how and why it's being used. I agree that its ability to trace the entire route a request took through every service is really impressive and is a _huge differentiator_ from other profiling tools, but from what I can tell it's a profiling tool that also traces rather than a tracing tool that also profiles.

## Proposal

What I would propose is that the verbiage be updated from "Jaeger: open source, end-to-end distributed tracing" to be something to the effect of:

> Jaeger: open source, end-to-end distributed microservices profiling
(using your own terse elevator-pitch style)

Or maybe...

> Jaeger: An open source profiler which traces the entire route of requests through all microservices, providing an end-to-end perspective.
(using my own slightly more verbose style)

## Second Take: Make it True?

Another option is to simply make it true. Although it is a tall order, if you were to extend Jaeger to be a logging utility as well, then had the ability to enable/disable trace logging entries within samples I see this as a huge boon to the value of this software. Even then, it still is a darn good profiler so I would still call it both. Maybe something like this:

> Jaeger: open source, end-to-end distributed microservices profiling & tracing

## Recap - Why This Matters

Again, I don't think myself or anyone else really cares that much what you call yourself or how you brand on an emotional level, but on a practical level I believe this simple change would have had the following impact:

- The true purpose and value of the tool would have been understood with no need to search for it or ask more questions
- Our team wouldn't have wasted so much time trying to figure out what the tool does
- We wouldn't have kept asking how to use it as a logging library to log traces 

That said, maybe I still have this wrong and I'm missing the primary use case/value proposition of this tool? I surely hope not, but if I've managed to make it this far and still not understand then any help on my journey would be greatly appreciated.

**Comment by yurishkuro** [comment]
Thank you for the feedback. There is definitely a lot of room on the website to describe the purpose of the tool better to newcomers. Distributed tracing is a relatively well-established term in the industry, but it is not necessarily the most self-descriptive, and in fact the concept is not the easiest to explain (I've had similar discussions at KubeCons).

I wouldn't want to change the actual tag line, because "distributed tracing platform" describes exactly what Jaeger is. But the website can explain, in the follow-up paragraphs, how Jaeger can be used.

I don't agree with the use of the word "profiler". While it is certainly relevant, it's an attempt to explain distributed tracing by analogy, instead of explaining it directly. Profiling has a fairly specific meaning to engineers, and distributed tracing is substantially different, as well as much broader in the use cases that it covers. For example, I don't think anyone would call the task of _root cause isolation_ (e.g., where in my distributed stack the error originates) a profiling activity.

FWIW, I have not seen a very precise, one-sentence definition of distributed tracing in 7yrs working in this space.

**Reply by ChadBailey** [reply]
[transferred from individual comment to this reply]

Thanks so much for your quick response. As luck would have it, although I waited to post this until I learned what I consider a reasonable amount, I think I have discovered the primary disconnect not long after posting it.

Jaeger seems to be generating the raw data, while other tools consume that data to be turned into something more useful (such as jaeger-ui). I think our discussions may have been around jaeger-ui specifically, which was a big part of what lead to the confusion.

If the previous statement is correct I think I would consider jaeger a logging client, but concede calling it logging would carry many of the same challenges as calling it tracing does today if not more. That said I think what's specifically being called a "trace" in this space may have benefitted from having its own newly coined term or perhaps using something more generic like a "frame"/"data frame".

I certainly don't have the answer, and am not suggesting you accept my half-baked thoughts as the answer, just socializing my thought process. Often times I think the perspective of someone totally fresh to an area of discipline can see things that others who have been there a long time can't, so I will keep thinking on this as I learn and update if I can formulate my thoughts further.

At the end of the day, the problem I'm interested in solving is: "How can you quickly communicate the intent and value of jaeger to software professionals who aren't specialized in monitoring?" If I can contribute to solving that in some way, I'd be thrilled.

I'll keep thinking on it and report back here if I come up with anything. More to come (hopefully)...

**Comment by ChadBailey** [comment]
[transferred to reply](https://github.com/jaegertracing/jaeger/discussions/3758#discussioncomment-2958254)

**Reply by yurishkuro** [reply]
I highly recommend this paper as an introduction to tracing space: https://www.pdl.cmu.edu/ftp/SelfStar/CMU-PDL-14-102.pdf

**Reply by ChadBailey** [reply]
Thanks so much! Sorry, it just occurred to me that I should have written this as a nested reply rather than a new comment. Still learning github discussions too apparently 😟
